src/java.base/share/classes/sun/net/InetAddressCachePolicy.java

```diff
@@ -121,40 +112,61 @@ public Integer run() {
                 cachePolicy = DEFAULT_POSITIVE;
             }
         }
-        tmp = java.security.AccessController.doPrivileged (
-          new PrivilegedAction<Integer>() {
-            public Integer run() {
-                try {
-                    String tmpString = Security.getProperty(negativeCachePolicyProp);
-                    if (tmpString != null) {
-                        return Integer.valueOf(tmpString);
-                    }
-                } catch (NumberFormatException ignored) {
-                    // Ignore
-                }
-
-                try {
-                    String tmpString = System.getProperty(negativeCachePolicyPropFallback);
-                    if (tmpString != null) {
-                        return Integer.decode(tmpString);
-                    }
-                } catch (NumberFormatException ignored) {
-                    // Ignore
-                }
-                return null;
-            }
-          });
+        tmp = getProperty(negativeCachePolicyProp,
+                          negativeCachePolicyPropFallback);
 
         if (tmp != null) {
             negativeCachePolicy = tmp < 0 ? FOREVER : tmp;
             propertyNegativeSet = true;
         }
+        long max = TimeUnit.DAYS.toSeconds(7);
+        if (cachePolicy > 0 && cachePolicy < max) {
+            tmp = getProperty(cacheStalePolicyProp,
+                              cacheStalePolicyPropFallback);
+            if (tmp != null && tmp > cachePolicy) {
```

Comment:
Why do you require cacheStalePolicy > cachePolicy? In your implementation, stale timer starts running after the base timer expires, so lower values still make sense.